src/simulation_strategies/FixedLossSimulationStrategy.py

```diff
@@ -46,35 +47,24 @@ def __randomize_modes_phases(state_in_modes_basis: ndarray) -> ndarray:
             :param state_in_modes_basis: A given state in modes basis.
             :return: Given mode state with randomized phases.
         """
-        randomized_phases_state = []
-
-        for mode in state_in_modes_basis:
-            phi = random() * 2 * pi
-            randomized_phases_state.append(exp(1j * phi) * mode)
-
+        randomized_phases_state = exp(1j * rand(len(state_in_modes_basis))) * state_in_modes_basis
         return randomized_phases_state
 
     @staticmethod
     def __calculate_probabilities(state: ndarray) -> ndarray:
-        probabilities = []
-        for detector in state:
-            probabilities.append(conjugate(detector) * detector)
+        probabilities = [conjugate(detector) * detector for detector in state]
         return probabilities
 
     def __calculate_approximation_of_boson_sampling_outcome(self, probabilities: ndarray) -> ndarray:
         """
             This method applies evolution to every photon. Note, that evolution of each particle is independent of
             each other.
             :param probabilities:
-            :return:
+            :return: An output array that contains the number of observations for each photon left (histogram)
         """
```

Comment:
Have a good day @Tomev 
In this case, the behaviour was designed as an histogram because of this code equivalency.
I am happy to contribute to the project.
_Please_, could you check it if the equivalency is right?